2025-02-07 20:51:20,860 - ERROR - Bitget get_price HTTP error: 429 for BNBUSDT_SPBL
2025-02-07 20:51:20,864 - ERROR - Bitget get_price HTTP error: 429 for COMBOUSDT_SPBL
2025-02-07 20:51:20,867 - ERROR - Bitget get_price HTTP error: 429 for FDUSDUSDT_SPBL
2025-02-07 20:51:20,953 - ERROR - Bitget get_price HTTP error: 429 for STRKUSDT_SPBL
2025-02-07 20:51:20,990 - ERROR - Bitget get_price HTTP error: 429 for DODOUSDT_SPBL
2025-02-07 20:51:21,096 - ERROR - Bitget get_price HTTP error: 429 for REZUSDT_SPBL
2025-02-07 20:51:21,102 - ERROR - Bitget get_price HTTP error: 429 for ACTUSDT_SPBL
2025-02-07 20:51:21,105 - ERROR - Bitget get_price HTTP error: 429 for CGPTUSDT_SPBL
2025-02-07 20:51:21,117 - ERROR - Bitget get_price HTTP error: 429 for SNTUSDT_SPBL
2025-02-07 20:51:21,145 - ERROR - Bitget get_price HTTP error: 429 for SUPERUSDT_SPBL
2025-02-07 20:51:21,156 - ERROR - Bitget get_price HTTP error: 429 for ETHUSDT_SPBL
2025-02-07 20:51:21,172 - ERROR - Bitget get_price HTTP error: 429 for LPTUSDT_SPBL
2025-02-07 20:51:21,178 - ERROR - Bitget get_price HTTP error: 429 for OMUSDT_SPBL
2025-02-07 20:51:21,181 - ERROR - Bitget get_price HTTP error: 429 for IQUSDT_SPBL
2025-02-07 20:51:21,184 - ERROR - Bitget get_price HTTP error: 429 for ZKUSDT_SPBL
2025-02-07 20:51:21,191 - ERROR - Bitget get_price HTTP error: 429 for AAVEBTC_SPBL
2025-02-07 20:51:21,197 - ERROR - Bitget get_price HTTP error: 429 for NEXOUSDT_SPBL
2025-02-07 20:51:21,200 - ERROR - Bitget get_price HTTP error: 429 for BTCBRL_SPBL
2025-02-07 20:51:21,203 - ERROR - Bitget get_price HTTP error: 429 for TRXETH_SPBL
2025-02-07 20:51:21,206 - ERROR - Bitget get_price HTTP error: 429 for ETHFIUSDT_SPBL
2025-02-07 20:51:21,255 - ERROR - Bitget get_price HTTP error: 429 for ICXUSDT_SPBL
2025-02-07 20:51:21,257 - ERROR - Bitget get_price HTTP error: 429 for ATMUSDT_SPBL
2025-02-07 20:51:21,294 - ERROR - Bitget get_price HTTP error: 429 for BCHUSDT_SPBL
2025-02-07 20:51:21,299 - ERROR - Bitget get_price HTTP error: 429 for JUPUSDT_SPBL
2025-02-07 20:51:21,301 - ERROR - Bitget get_price HTTP error: 429 for NOTUSDT_SPBL
2025-02-07 20:51:21,304 - ERROR - Bitget get_price HTTP error: 429 for GRTUSDT_SPBL
2025-02-07 20:51:21,307 - ERROR - Bitget get_price HTTP error: 429 for IOUSDT_SPBL
2025-02-07 20:51:21,310 - ERROR - Bitget get_price HTTP error: 429 for AAVEUSDC_SPBL
2025-02-07 20:51:21,314 - ERROR - Bitget get_price HTTP error: 429 for LUNCUSDT_SPBL
2025-02-07 20:51:21,317 - ERROR - Bitget get_price HTTP error: 429 for LINKBTC_SPBL
2025-02-07 20:51:21,319 - ERROR - Bitget get_price HTTP error: 429 for VICUSDT_SPBL
2025-02-07 20:51:21,911 - ERROR - Bitget get_price HTTP error: 429 for AUCTIONUSDT_SPBL
2025-02-07 20:51:21,950 - ERROR - Bitget get_price HTTP error: 429 for STXUSDT_SPBL
2025-02-07 20:51:21,957 - ERROR - Bitget get_price HTTP error: 429 for DOTUSDT_SPBL
2025-02-07 20:51:21,959 - ERROR - Bitget get_price HTTP error: 429 for ACAUSDT_SPBL
2025-02-07 20:51:22,071 - ERROR - Bitget get_price HTTP error: 429 for HFTUSDT_SPBL
2025-02-07 20:51:22,074 - ERROR - Bitget get_price HTTP error: 429 for IDEXUSDT_SPBL
2025-02-07 20:51:22,077 - ERROR - Bitget get_price HTTP error: 429 for FILETH_SPBL
2025-02-07 20:51:22,102 - ERROR - Bitget get_price HTTP error: 429 for BIOUSDT_SPBL
2025-02-07 20:51:22,105 - ERROR - Bitget get_price HTTP error: 429 for COWUSDT_SPBL
2025-02-07 20:51:22,107 - ERROR - Bitget get_price HTTP error: 429 for ALPINEUSDT_SPBL
2025-02-07 20:51:22,109 - ERROR - Bitget get_price HTTP error: 429 for XRPEUR_SPBL
2025-02-07 20:51:22,117 - ERROR - Bitget get_price HTTP error: 429 for WIFEUR_SPBL
2025-02-07 20:51:22,121 - ERROR - Bitget get_price HTTP error: 429 for WINGUSDT_SPBL
2025-02-07 20:51:22,192 - ERROR - Bitget get_price HTTP error: 429 for MANAUSDT_SPBL
2025-02-07 20:51:22,196 - ERROR - Bitget get_price HTTP error: 429 for ETCUSDT_SPBL
2025-02-07 20:51:22,201 - ERROR - Bitget get_price HTTP error: 429 for PHAUSDT_SPBL
2025-02-07 20:51:22,203 - ERROR - Bitget get_price HTTP error: 429 for GHSTUSDT_SPBL
2025-02-07 20:51:22,210 - ERROR - Bitget get_price HTTP error: 429 for ETHEUR_SPBL
2025-02-07 20:51:22,216 - ERROR - Bitget get_price HTTP error: 429 for ALPHAUSDT_SPBL
2025-02-07 20:51:22,234 - ERROR - Bitget get_price HTTP error: 429 for AVAXUSDT_SPBL
2025-02-07 20:51:22,237 - ERROR - Bitget get_price HTTP error: 429 for STMXUSDT_SPBL
2025-02-07 20:51:22,241 - ERROR - Bitget get_price HTTP error: 429 for DOTUSDC_SPBL
2025-02-07 20:51:22,262 - ERROR - Bitget get_price HTTP error: 429 for HIVEUSDT_SPBL
2025-02-07 20:51:22,275 - ERROR - Bitget get_price HTTP error: 429 for MKRUSDT_SPBL
2025-02-07 20:51:22,282 - ERROR - Bitget get_price HTTP error: 429 for GLMRUSDT_SPBL
2025-02-07 20:51:22,287 - ERROR - Bitget get_price HTTP error: 429 for GMTUSDT_SPBL
2025-02-07 20:51:22,312 - ERROR - Bitget get_price HTTP error: 429 for SHIBUSDT_SPBL
2025-02-07 20:51:22,348 - ERROR - Bitget get_price HTTP error: 429 for JSTUSDT_SPBL
2025-02-07 20:51:22,360 - ERROR - Bitget get_price HTTP error: 429 for WINUSDT_SPBL
2025-02-07 20:51:22,371 - ERROR - Bitget get_price HTTP error: 429 for ENAUSDT_SPBL
2025-02-07 20:51:22,441 - ERROR - Bitget get_price HTTP error: 429 for ADAEUR_SPBL
2025-02-07 20:51:22,801 - ERROR - Bitget get_price HTTP error: 429 for ACHUSDT_SPBL
2025-02-07 20:51:22,910 - ERROR - Bitget get_price HTTP error: 429 for TRUMPUSDT_SPBL
2025-02-07 20:51:22,921 - ERROR - Bitget get_price HTTP error: 429 for LTCETH_SPBL
2025-02-07 20:51:22,953 - ERROR - Bitget get_price HTTP error: 429 for ANKRUSDT_SPBL
2025-02-07 20:51:22,980 - ERROR - Bitget get_price HTTP error: 429 for AIXBTUSDT_SPBL
2025-02-07 20:51:22,986 - ERROR - Bitget get_price HTTP error: 429 for UNIUSDC_SPBL
2025-02-07 20:51:23,043 - ERROR - Bitget get_price HTTP error: 429 for AGLDUSDT_SPBL
2025-02-07 20:51:23,091 - ERROR - Bitget get_price HTTP error: 429 for SUNUSDT_SPBL
2025-02-07 20:51:23,094 - ERROR - Bitget get_price HTTP error: 429 for INJUSDC_SPBL
2025-02-07 20:51:23,106 - ERROR - Bitget get_price HTTP error: 429 for LRCUSDT_SPBL
2025-02-07 20:51:23,111 - ERROR - Bitget get_price HTTP error: 429 for DENTUSDT_SPBL
2025-02-07 20:51:23,114 - ERROR - Bitget get_price HTTP error: 429 for ONEUSDT_SPBL
2025-02-07 20:51:23,121 - ERROR - Bitget get_price HTTP error: 429 for LINKEUR_SPBL
2025-02-07 20:51:23,130 - ERROR - Bitget get_price HTTP error: 429 for LITUSDT_SPBL
2025-02-07 20:51:23,153 - ERROR - Bitget get_price HTTP error: 429 for NEARUSDC_SPBL
2025-02-07 20:51:23,161 - ERROR - Bitget get_price HTTP error: 429 for INJUSDT_SPBL
2025-02-07 20:51:23,163 - ERROR - Bitget get_price HTTP error: 429 for WLDEUR_SPBL
2025-02-07 20:51:23,169 - ERROR - Bitget get_price HTTP error: 429 for USDCUSDT_SPBL
2025-02-07 20:51:23,173 - ERROR - Bitget get_price HTTP error: 429 for AAVEUSDT_SPBL
2025-02-07 20:51:23,205 - ERROR - Bitget get_price HTTP error: 429 for NFPUSDT_SPBL
2025-02-07 20:51:23,207 - ERROR - Bitget get_price HTTP error: 429 for CKBUSDT_SPBL
2025-02-07 20:51:23,213 - ERROR - Bitget get_price HTTP error: 429 for AVAXUSDC_SPBL
2025-02-07 20:51:23,216 - ERROR - Bitget get_price HTTP error: 429 for PHAUSDC_SPBL
2025-02-07 20:51:23,220 - ERROR - Bitget get_price HTTP error: 429 for GNOUSDT_SPBL
2025-02-07 20:51:23,222 - ERROR - Bitget get_price HTTP error: 429 for ARBUSDC_SPBL
2025-02-07 20:51:23,224 - ERROR - Bitget get_price HTTP error: 429 for PENDLEUSDT_SPBL
2025-02-07 20:51:23,227 - ERROR - Bitget get_price HTTP error: 429 for EIGENUSDT_SPBL
2025-02-07 20:51:23,231 - ERROR - Bitget get_price HTTP error: 429 for FILUSDT_SPBL
2025-02-07 20:51:23,245 - ERROR - Bitget get_price HTTP error: 429 for HIFIUSDT_SPBL
2025-02-07 20:51:23,274 - ERROR - Bitget get_price HTTP error: 429 for FLMUSDT_SPBL
2025-02-07 20:51:23,290 - ERROR - Bitget get_price HTTP error: 429 for JTOUSDT_SPBL
2025-02-07 20:51:23,745 - ERROR - Bitget get_price HTTP error: 429 for LISTAUSDT_SPBL
2025-02-07 20:51:26,280 - ERROR - Kucoin get_price error for MAN-TAUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:26,301 - ERROR - Kucoin get_price error for ANI-MEUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:26,323 - ERROR - Kucoin get_price error for ARB-TC: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:26,337 - ERROR - Kucoin get_price error for DAS-HUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:26,344 - ERROR - Kucoin get_price error for COM-BOUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:26,358 - ERROR - Kucoin get_price error for DOD-OUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:26,370 - ERROR - Kucoin get_price error for ALI-CEUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:26,409 - ERROR - Kucoin get_price error for COO-KIEUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:26,419 - ERROR - Kucoin get_price error for OPU-SDC: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:26,439 - ERROR - Kucoin get_price error for NTR-NUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:26,472 - ERROR - Kucoin get_price error for USD-TBRL: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:26,486 - ERROR - Kucoin get_price error for HBA-RUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:26,493 - ERROR - Kucoin get_price error for RUN-EUSDC: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:26,504 - ERROR - Kucoin get_price error for HBA-RBTC: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:26,512 - ERROR - Kucoin get_price error for DAS-HBTC: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:26,521 - ERROR - Kucoin get_price error for FID-AUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:26,532 - ERROR - Kucoin get_price error for CAK-EUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:26,543 - ERROR - Kucoin get_price error for VAN-RYUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:26,555 - ERROR - Kucoin get_price error for DEG-OUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:26,569 - ERROR - Kucoin get_price error for USD-TDAI: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:26,579 - ERROR - Kucoin get_price error for ORD-IUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:26,616 - ERROR - Kucoin get_price error for FLO-KIUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:26,691 - ERROR - Kucoin get_price error for PAX-GUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:26,698 - ERROR - Kucoin get_price error for REN-DERUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:26,718 - ERROR - Kucoin get_price error for IOT-ABTC: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:26,724 - ERROR - Kucoin get_price error for ORC-AUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:26,732 - ERROR - Kucoin get_price error for FLO-WUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:26,739 - ERROR - Kucoin get_price error for OSM-OUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:26,753 - ERROR - Kucoin get_price error for AER-GOUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:26,772 - ERROR - Kucoin get_price error for BUR-GERUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:26,789 - ERROR - Kucoin get_price error for IOT-XETH: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:26,806 - ERROR - Kucoin get_price error for DOG-SUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:26,814 - ERROR - Kucoin get_price error for SUS-DT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:26,823 - ERROR - Kucoin get_price error for MIN-AUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:26,832 - ERROR - Kucoin get_price error for PER-PBTC: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:26,878 - ERROR - Kucoin get_price error for CGP-TUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:26,960 - ERROR - Kucoin get_price error for HMS-TRUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:26,970 - ERROR - Kucoin get_price error for ANK-RBTC: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:26,991 - ERROR - Kucoin get_price error for ZKU-SDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:27,000 - ERROR - Kucoin get_price error for SUP-ERBTC: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:27,011 - ERROR - Kucoin get_price error for AAV-EBTC: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:27,114 - ERROR - Kucoin get_price error for BON-KUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:27,123 - ERROR - Kucoin get_price error for OMU-SDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:27,136 - ERROR - Kucoin get_price error for CYB-ERUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:27,141 - ERROR - Kucoin get_price error for ETH-FIUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:27,146 - ERROR - Kucoin get_price error for TNS-RUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:27,162 - ERROR - Kucoin get_price error for WBT-CBTC: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:27,174 - ERROR - Kucoin get_price error for LIN-KBTC: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:27,182 - ERROR - Kucoin get_price error for IOU-SDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:27,191 - ERROR - Kucoin get_price error for VAN-AUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:27,201 - ERROR - Kucoin get_price error for SUS-DC: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:27,207 - ERROR - Kucoin get_price error for LUN-CUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:27,214 - ERROR - Kucoin get_price error for ALP-HABTC: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:27,222 - ERROR - Kucoin get_price error for LIN-KUSDC: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:27,293 - ERROR - Kucoin get_price error for ALG-OBTC: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:27,304 - ERROR - Kucoin get_price error for MAG-ICUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:27,310 - ERROR - Kucoin get_price error for IOT-AUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:27,362 - ERROR - Kucoin get_price error for DOG-EUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:27,374 - ERROR - Kucoin get_price error for NEI-ROUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:27,386 - ERROR - Kucoin get_price error for STR-KUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:27,403 - ERROR - Kucoin get_price error for TUS-DUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:27,424 - ERROR - Kucoin get_price error for SCR-TBTC: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:27,430 - ERROR - Kucoin get_price error for SUP-ERUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:27,436 - ERROR - Kucoin get_price error for IOT-XBTC: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:27,443 - ERROR - Kucoin get_price error for DOG-EBTC: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:27,481 - ERROR - Kucoin get_price error for SCR-TUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:27,496 - ERROR - Kucoin get_price error for DEX-EUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:27,522 - ERROR - Kucoin get_price error for JAS-MYUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:27,533 - ERROR - Kucoin get_price error for SAN-DUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:27,540 - ERROR - Kucoin get_price error for AEV-OUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:27,555 - ERROR - Kucoin get_price error for MOV-RUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:27,590 - ERROR - Kucoin get_price error for NEA-RBTC: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:27,597 - ERROR - Kucoin get_price error for AUD-IOUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:27,606 - ERROR - Kucoin get_price error for ARP-AUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:27,613 - ERROR - Kucoin get_price error for BIC-OUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:27,636 - ERROR - Kucoin get_price error for ARU-SDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:27,642 - ERROR - Kucoin get_price error for CEL-RUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:27,650 - ERROR - Kucoin get_price error for PEO-PLEUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:27,656 - ERROR - Kucoin get_price error for PUN-DIXUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:27,668 - ERROR - Kucoin get_price error for DUS-KUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:27,695 - ERROR - Kucoin get_price error for DAS-HETH: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:27,748 - ERROR - Kucoin get_price error for IDU-SDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:27,802 - ERROR - Kucoin get_price error for STO-RJUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:27,809 - ERROR - Kucoin get_price error for CEL-OBTC: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:27,826 - ERROR - Kucoin get_price error for SUS-HIUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:27,841 - ERROR - Kucoin get_price error for FLO-WBTC: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:27,865 - ERROR - Kucoin get_price error for AVA-XBTC: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:27,906 - ERROR - Kucoin get_price error for LIN-AUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:27,919 - ERROR - Kucoin get_price error for PYT-HUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:27,937 - ERROR - Kucoin get_price error for CEL-OUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:28,004 - ERROR - Kucoin get_price error for IOT-XUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:28,014 - ERROR - Kucoin get_price error for VOX-ELUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:28,021 - ERROR - Kucoin get_price error for GLM-RBTC: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:28,026 - ERROR - Kucoin get_price error for SHI-BUSDC: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:28,042 - ERROR - Kucoin get_price error for AUC-TIONUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:28,148 - ERROR - Kucoin get_price error for CRE-AMUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:28,206 - ERROR - Kucoin get_price error for STR-AXUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:28,216 - ERROR - Kucoin get_price error for USU-ALUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:28,225 - ERROR - Kucoin get_price error for QUI-CKUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:28,268 - ERROR - Kucoin get_price error for COM-PUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:28,275 - ERROR - Kucoin get_price error for PEP-EUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:28,286 - ERROR - Kucoin get_price error for THE-TAUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:28,294 - ERROR - Kucoin get_price error for CTS-IUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:28,304 - ERROR - Kucoin get_price error for ALP-INEUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:28,310 - ERROR - Kucoin get_price error for PON-DBTC: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:28,322 - ERROR - Kucoin get_price error for TFU-ELBTC: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:28,369 - ERROR - Kucoin get_price error for AST-RBTC: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:28,374 - ERROR - Kucoin get_price error for ARK-MUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:28,385 - ERROR - Kucoin get_price error for PAX-GBTC: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:28,508 - ERROR - Kucoin get_price error for ATO-MUSDC: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:28,515 - ERROR - Kucoin get_price error for DAT-AUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:28,525 - ERROR - Kucoin get_price error for AVA-XUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:28,581 - ERROR - Kucoin get_price error for TFU-ELUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:28,586 - ERROR - Kucoin get_price error for MAN-AUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:28,593 - ERROR - Kucoin get_price error for FOR-THUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:28,602 - ERROR - Kucoin get_price error for GLM-RUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:28,608 - ERROR - Kucoin get_price error for OMB-TC: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:28,618 - ERROR - Kucoin get_price error for SHI-BDOGE: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:28,645 - ERROR - Kucoin get_price error for OPU-SDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:28,668 - ERROR - Kucoin get_price error for SHI-BUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:28,687 - ERROR - Kucoin get_price error for BAN-ANAUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:28,710 - ERROR - Kucoin get_price error for PRO-MUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:28,725 - ERROR - Kucoin get_price error for TUS-DT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:28,765 - ERROR - Kucoin get_price error for ROS-EUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:28,769 - ERROR - Kucoin get_price error for LQT-YUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:28,774 - ERROR - Kucoin get_price error for PNU-TUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:28,791 - ERROR - Kucoin get_price error for BER-AUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:28,809 - ERROR - Kucoin get_price error for ALI-CEBTC: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:28,815 - ERROR - Kucoin get_price error for LIN-KUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:28,822 - ERROR - Kucoin get_price error for SOL-VUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:28,871 - ERROR - Kucoin get_price error for PON-DUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:28,889 - ERROR - Kucoin get_price error for WBT-CUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:28,908 - ERROR - Kucoin get_price error for ATO-MBTC: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:28,964 - ERROR - Kucoin get_price error for 1IN-CHUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:28,976 - ERROR - Kucoin get_price error for FLO-KIUSDC: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:29,026 - ERROR - Kucoin get_price error for KAI-AUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:29,049 - ERROR - Kucoin get_price error for COT-IUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:29,061 - ERROR - Kucoin get_price error for MAN-ABTC: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:29,122 - ERROR - Kucoin get_price error for WUS-DT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:29,127 - ERROR - Kucoin get_price error for RDN-TUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:29,142 - ERROR - Kucoin get_price error for RUN-EUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:29,173 - ERROR - Kucoin get_price error for ALP-HAUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:29,181 - ERROR - Kucoin get_price error for ATO-MUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:29,187 - ERROR - Kucoin get_price error for MET-ISUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:29,202 - ERROR - Kucoin get_price error for TRU-MPUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:29,209 - ERROR - Kucoin get_price error for BOM-EUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:29,216 - ERROR - Kucoin get_price error for LUN-AUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:29,250 - ERROR - Kucoin get_price error for RUN-EBTC: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:29,260 - ERROR - Kucoin get_price error for HIG-HUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:29,287 - ERROR - Kucoin get_price error for DAT-ABTC: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:29,293 - ERROR - Kucoin get_price error for ANK-RUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:29,311 - ERROR - Kucoin get_price error for ALG-OUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:29,335 - ERROR - Kucoin get_price error for AIX-BTUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:29,340 - ERROR - Kucoin get_price error for PEN-GUUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:29,346 - ERROR - Kucoin get_price error for AGL-DUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:29,352 - ERROR - Kucoin get_price error for NEA-RUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:29,367 - ERROR - Kucoin get_price error for EGL-DBTC: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:29,401 - ERROR - Kucoin get_price error for DEN-TUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:29,406 - ERROR - Kucoin get_price error for QIU-SDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:29,418 - ERROR - Kucoin get_price error for ATO-METH: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:29,433 - ERROR - Kucoin get_price error for USD-PUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:29,439 - ERROR - Kucoin get_price error for AAV-EUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:29,483 - ERROR - Kucoin get_price error for NEA-RUSDC: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:29,487 - ERROR - Kucoin get_price error for CTS-IBTC: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:29,493 - ERROR - Kucoin get_price error for USD-CUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:29,508 - ERROR - Kucoin get_price error for EIG-ENUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:29,523 - ERROR - Kucoin get_price error for PEN-DLEUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:29,534 - ERROR - Kucoin get_price error for GUS-DT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:29,539 - ERROR - Kucoin get_price error for HIF-IUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:29,559 - ERROR - Kucoin get_price error for AVA-XUSDC: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:29,580 - ERROR - Kucoin get_price error for PIX-ELUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:29,612 - ERROR - Kucoin get_price error for DEN-TETH: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:29,618 - ERROR - Kucoin get_price error for POL-YXUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:29,634 - ERROR - Kucoin get_price error for UST-CUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:29,640 - ERROR - Kucoin get_price error for HAR-DUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:29,647 - ERROR - Kucoin get_price error for BAN-DBTC: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:29,671 - ERROR - Kucoin get_price error for ALG-OUSDC: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:29,677 - ERROR - Kucoin get_price error for BAN-DUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:29,683 - ERROR - Kucoin get_price error for VID-TUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:29,721 - ERROR - Kucoin get_price error for BBU-SDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:29,736 - ERROR - Kucoin get_price error for AUD-IOBTC: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:29,743 - ERROR - Kucoin get_price error for STO-RJBTC: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:29,754 - ERROR - Kucoin get_price error for CAT-IUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:29,786 - ERROR - Kucoin get_price error for QTU-MUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:29,793 - ERROR - Kucoin get_price error for BIF-IUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:29,812 - ERROR - Kucoin get_price error for DYD-XUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:29,846 - ERROR - Kucoin get_price error for MAS-KUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:29,851 - ERROR - Kucoin get_price error for MOV-EUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:29,857 - ERROR - Kucoin get_price error for MAN-AETH: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:29,874 - ERROR - Kucoin get_price error for POR-TALUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:29,920 - ERROR - Kucoin get_price error for LOK-AUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:29,930 - ERROR - Kucoin get_price error for BLU-RUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:29,936 - ERROR - Kucoin get_price error for PER-PUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:29,944 - ERROR - Kucoin get_price error for OMN-IUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:29,997 - ERROR - Kucoin get_price error for API-3USDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:30,003 - ERROR - Kucoin get_price error for EGL-DUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:30,009 - ERROR - Kucoin get_price error for COT-IBTC: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:30,023 - ERROR - Kucoin get_price error for VTH-OUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:30,039 - ERROR - Kucoin get_price error for LIS-TAUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:30,056 - ERROR - Kucoin get_price error for DUS-DT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:30,064 - ERROR - Kucoin get_price error for LUM-IAUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:30,080 - ERROR - Kucoin get_price error for IOS-TUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:30,102 - ERROR - Kucoin get_price error for AST-RUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:30,112 - ERROR - Kucoin get_price error for CET-USUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:30,120 - ERROR - Kucoin get_price error for MEU-SDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:30,130 - ERROR - Kucoin get_price error for DOG-EUSDC: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:30,137 - ERROR - Kucoin get_price error for MEM-EUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:30,147 - ERROR - Kucoin get_price error for FLU-XUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:30,160 - ERROR - Kucoin get_price error for KAV-AUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:30,166 - ERROR - Kucoin get_price error for TUR-BOUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:41,712 - ERROR - Bitget get_price HTTP error: 429 for BNBUSDT_SPBL
2025-02-07 20:51:41,772 - ERROR - Bitget get_price HTTP error: 429 for LTCEUR_SPBL
2025-02-07 20:51:41,774 - ERROR - Bitget get_price HTTP error: 429 for HBARUSDT_SPBL
2025-02-07 20:51:41,787 - ERROR - Bitget get_price HTTP error: 429 for ACXUSDT_SPBL
2025-02-07 20:51:41,788 - ERROR - Bitget get_price HTTP error: 429 for IMXUSDT_SPBL
2025-02-07 20:51:41,810 - ERROR - Bitget get_price HTTP error: 429 for CAKEUSDT_SPBL
2025-02-07 20:51:41,987 - ERROR - Bitget get_price HTTP error: 429 for SNTUSDT_SPBL
2025-02-07 20:51:41,991 - ERROR - Bitget get_price HTTP error: 429 for SUPERUSDT_SPBL
2025-02-07 20:51:41,997 - ERROR - Bitget get_price HTTP error: 429 for ICXUSDT_SPBL
2025-02-07 20:51:42,008 - ERROR - Bitget get_price HTTP error: 429 for TKOUSDT_SPBL
2025-02-07 20:51:42,026 - ERROR - Bitget get_price HTTP error: 429 for UNIUSDT_SPBL
2025-02-07 20:51:42,034 - ERROR - Bitget get_price HTTP error: 429 for HMSTRUSDT_SPBL
2025-02-07 20:51:42,038 - ERROR - Bitget get_price HTTP error: 429 for ZKUSDT_SPBL
2025-02-07 20:51:42,041 - ERROR - Bitget get_price HTTP error: 429 for ETHUSDT_SPBL
2025-02-07 20:51:42,044 - ERROR - Bitget get_price HTTP error: 429 for SUIUSDC_SPBL
2025-02-07 20:51:42,073 - ERROR - Bitget get_price HTTP error: 429 for OXTUSDT_SPBL
2025-02-07 20:51:42,075 - ERROR - Bitget get_price HTTP error: 429 for NEXOUSDT_SPBL
2025-02-07 20:51:42,077 - ERROR - Bitget get_price HTTP error: 429 for OMUSDT_SPBL
2025-02-07 20:51:42,081 - ERROR - Bitget get_price HTTP error: 429 for TNSRUSDT_SPBL
2025-02-07 20:51:42,083 - ERROR - Bitget get_price HTTP error: 429 for AAVEBTC_SPBL
2025-02-07 20:51:42,088 - ERROR - Bitget get_price HTTP error: 429 for BTCBRL_SPBL
2025-02-07 20:51:42,090 - ERROR - Bitget get_price HTTP error: 429 for SCRUSDT_SPBL
2025-02-07 20:51:42,105 - ERROR - Bitget get_price HTTP error: 429 for CYBERUSDT_SPBL
2025-02-07 20:51:42,144 - ERROR - Bitget get_price HTTP error: 429 for IQUSDT_SPBL
2025-02-07 20:51:42,145 - ERROR - Bitget get_price HTTP error: 429 for LEVERUSDT_SPBL
2025-02-07 20:51:42,148 - ERROR - Bitget get_price HTTP error: 429 for ELFUSDT_SPBL
2025-02-07 20:51:42,156 - ERROR - Bitget get_price HTTP error: 429 for BAKEUSDT_SPBL
2025-02-07 20:51:42,158 - ERROR - Bitget get_price HTTP error: 429 for LUNCUSDT_SPBL
2025-02-07 20:51:42,160 - ERROR - Bitget get_price HTTP error: 429 for FETUSDT_SPBL
2025-02-07 20:51:42,162 - ERROR - Bitget get_price HTTP error: 429 for IOUSDT_SPBL
2025-02-07 20:51:42,163 - ERROR - Bitget get_price HTTP error: 429 for GRTUSDT_SPBL
2025-02-07 20:51:42,181 - ERROR - Bitget get_price HTTP error: 429 for BCHUSDT_SPBL
2025-02-07 20:51:42,215 - ERROR - Bitget get_price HTTP error: 429 for ARBUSDT_SPBL
2025-02-07 20:51:42,225 - ERROR - Bitget get_price HTTP error: 429 for FTTUSDT_SPBL
2025-02-07 20:51:42,227 - ERROR - Bitget get_price HTTP error: 429 for AAVEUSDC_SPBL
2025-02-07 20:51:42,228 - ERROR - Bitget get_price HTTP error: 429 for JUPUSDT_SPBL
2025-02-07 20:51:42,230 - ERROR - Bitget get_price HTTP error: 429 for LINKBTC_SPBL
2025-02-07 20:51:42,236 - ERROR - Bitget get_price HTTP error: 429 for SFPUSDT_SPBL
2025-02-07 20:51:42,239 - ERROR - Bitget get_price HTTP error: 429 for LINKUSDC_SPBL
2025-02-07 20:51:42,242 - ERROR - Bitget get_price HTTP error: 429 for ZRXUSDT_SPBL
2025-02-07 20:51:42,243 - ERROR - Bitget get_price HTTP error: 429 for SLFUSDT_SPBL
2025-02-07 20:51:42,270 - ERROR - Bitget get_price HTTP error: 429 for BCHUSDC_SPBL
2025-02-07 20:51:42,272 - ERROR - Bitget get_price HTTP error: 429 for DOGEUSDT_SPBL
2025-02-07 20:51:42,278 - ERROR - Bitget get_price HTTP error: 429 for MAGICUSDT_SPBL
2025-02-07 20:51:42,280 - ERROR - Bitget get_price HTTP error: 429 for WAXPUSDT_SPBL
2025-02-07 20:51:42,283 - ERROR - Bitget get_price HTTP error: 429 for BATUSDT_SPBL
2025-02-07 20:51:42,292 - ERROR - Bitget get_price HTTP error: 429 for AVAXEUR_SPBL
2025-02-07 20:51:42,294 - ERROR - Bitget get_price HTTP error: 429 for SOLUSDC_SPBL
2025-02-07 20:51:42,296 - ERROR - Bitget get_price HTTP error: 429 for MOVRUSDT_SPBL
2025-02-07 20:51:42,302 - ERROR - Bitget get_price HTTP error: 429 for TRXUSDC_SPBL
2025-02-07 20:51:42,305 - ERROR - Bitget get_price HTTP error: 429 for JASMYUSDT_SPBL
2025-02-07 20:51:42,389 - ERROR - Bitget get_price HTTP error: 429 for CHZUSDT_SPBL
2025-02-07 20:51:42,391 - ERROR - Bitget get_price HTTP error: 429 for SOLBTC_SPBL
2025-02-07 20:51:42,393 - ERROR - Bitget get_price HTTP error: 429 for ARUSDT_SPBL
2025-02-07 20:51:42,410 - ERROR - Bitget get_price HTTP error: 429 for AUDIOUSDT_SPBL
2025-02-07 20:51:42,412 - ERROR - Bitget get_price HTTP error: 429 for AEVOUSDT_SPBL
2025-02-07 20:51:42,414 - ERROR - Bitget get_price HTTP error: 429 for SEIUSDT_SPBL
2025-02-07 20:51:42,417 - ERROR - Bitget get_price HTTP error: 429 for BICOUSDT_SPBL
2025-02-07 20:51:42,421 - ERROR - Bitget get_price HTTP error: 429 for SANDUSDT_SPBL
2025-02-07 20:51:42,438 - ERROR - Bitget get_price HTTP error: 429 for PEOPLEUSDT_SPBL
2025-02-07 20:51:42,442 - ERROR - Bitget get_price HTTP error: 429 for CELRUSDT_SPBL
2025-02-07 20:51:42,445 - ERROR - Bitget get_price HTTP error: 429 for PYRUSDT_SPBL
2025-02-07 20:51:42,457 - ERROR - Bitget get_price HTTP error: 429 for PUNDIXUSDT_SPBL
2025-02-07 20:51:42,465 - ERROR - Bitget get_price HTTP error: 429 for STORJUSDT_SPBL
2025-02-07 20:51:42,470 - ERROR - Bitget get_price HTTP error: 429 for DOGEBTC_SPBL
2025-02-07 20:51:42,475 - ERROR - Bitget get_price HTTP error: 429 for XRPUSDT_SPBL
2025-02-07 20:51:42,490 - ERROR - Bitget get_price HTTP error: 429 for ILVUSDT_SPBL
2025-02-07 20:51:42,492 - ERROR - Bitget get_price HTTP error: 429 for DUSKUSDT_SPBL
2025-02-07 20:51:42,496 - ERROR - Bitget get_price HTTP error: 429 for IDUSDT_SPBL
2025-02-07 20:51:42,498 - ERROR - Bitget get_price HTTP error: 429 for ONGUSDT_SPBL
2025-02-07 20:51:42,526 - ERROR - Bitget get_price HTTP error: 429 for FETUSDC_SPBL
2025-02-07 20:51:42,541 - ERROR - Bitget get_price HTTP error: 429 for CHESSUSDT_SPBL
2025-02-07 20:51:42,544 - ERROR - Bitget get_price HTTP error: 429 for LAZIOUSDT_SPBL
2025-02-07 20:51:42,548 - ERROR - Bitget get_price HTTP error: 429 for LSKUSDT_SPBL
2025-02-07 20:51:42,584 - ERROR - Bitget get_price HTTP error: 429 for CFXUSDT_SPBL
2025-02-07 20:51:42,587 - ERROR - Bitget get_price HTTP error: 429 for FISUSDT_SPBL
2025-02-07 20:51:42,589 - ERROR - Bitget get_price HTTP error: 429 for BTCEUR_SPBL
2025-02-07 20:51:42,594 - ERROR - Bitget get_price HTTP error: 429 for PEPEUSDC_SPBL
2025-02-07 20:51:42,596 - ERROR - Bitget get_price HTTP error: 429 for AVAXBTC_SPBL
2025-02-07 20:51:42,599 - ERROR - Bitget get_price HTTP error: 429 for BELUSDT_SPBL
2025-02-07 20:51:42,602 - ERROR - Bitget get_price HTTP error: 429 for TLMUSDT_SPBL
2025-02-07 20:51:42,610 - ERROR - Bitget get_price HTTP error: 429 for XTZUSDT_SPBL
2025-02-07 20:51:42,617 - ERROR - Bitget get_price HTTP error: 429 for NKNUSDT_SPBL
2025-02-07 20:51:42,623 - ERROR - Bitget get_price HTTP error: 429 for LINAUSDT_SPBL
2025-02-07 20:51:42,650 - ERROR - Bitget get_price HTTP error: 429 for PYTHUSDT_SPBL
2025-02-07 20:51:42,709 - ERROR - Bitget get_price HTTP error: 429 for LTCUSDC_SPBL
2025-02-07 20:51:42,712 - ERROR - Bitget get_price HTTP error: 429 for ARPAUSDT_SPBL
2025-02-07 20:51:42,754 - ERROR - Bitget get_price HTTP error: 429 for SOLEUR_SPBL
2025-02-07 20:51:42,768 - ERROR - Bitget get_price HTTP error: 429 for IOTXUSDT_SPBL
2025-02-07 20:51:42,795 - ERROR - Bitget get_price HTTP error: 429 for BTCUSDT_SPBL
2025-02-07 20:51:43,340 - ERROR - Bitget get_price HTTP error: 429 for BERAUSDT_SPBL
2025-02-07 20:51:43,348 - ERROR - Bitget get_price HTTP error: 429 for SOLVUSDT_SPBL
2025-02-07 20:51:43,355 - ERROR - Bitget get_price HTTP error: 429 for ETHBTC_SPBL
2025-02-07 20:51:43,361 - ERROR - Bitget get_price HTTP error: 429 for PNUTUSDT_SPBL
2025-02-07 20:51:43,399 - ERROR - Bitget get_price HTTP error: 429 for RIFUSDT_SPBL
2025-02-07 20:51:43,401 - ERROR - Bitget get_price HTTP error: 429 for ATOMBTC_SPBL
2025-02-07 20:51:43,403 - ERROR - Bitget get_price HTTP error: 429 for ROSEUSDT_SPBL
2025-02-07 20:51:43,405 - ERROR - Bitget get_price HTTP error: 429 for FETBTC_SPBL
2025-02-07 20:51:43,426 - ERROR - Bitget get_price HTTP error: 429 for AMPUSDT_SPBL
2025-02-07 20:51:43,434 - ERROR - Bitget get_price HTTP error: 429 for POLUSDC_SPBL
2025-02-07 20:51:43,445 - ERROR - Bitget get_price HTTP error: 429 for LDOUSDT_SPBL
2025-02-07 20:51:43,457 - ERROR - Bitget get_price HTTP error: 429 for XLMUSDT_SPBL
2025-02-07 20:51:43,461 - ERROR - Bitget get_price HTTP error: 429 for WBTCUSDT_SPBL
2025-02-07 20:51:43,464 - ERROR - Bitget get_price HTTP error: 429 for FLOKIUSDC_SPBL
2025-02-07 20:51:43,468 - ERROR - Bitget get_price HTTP error: 429 for DYMUSDT_SPBL
2025-02-07 20:51:43,489 - ERROR - Bitget get_price HTTP error: 429 for TAOUSDC_SPBL
2025-02-07 20:51:43,495 - ERROR - Bitget get_price HTTP error: 429 for SKLUSDT_SPBL
2025-02-07 20:51:43,497 - ERROR - Bitget get_price HTTP error: 429 for 1INCHUSDT_SPBL
2025-02-07 20:51:43,500 - ERROR - Bitget get_price HTTP error: 429 for SUIUSDT_SPBL
2025-02-07 20:51:43,502 - ERROR - Bitget get_price HTTP error: 429 for MDTUSDT_SPBL
2025-02-07 20:51:43,504 - ERROR - Bitget get_price HTTP error: 429 for DOGEEUR_SPBL
2025-02-07 20:51:43,507 - ERROR - Bitget get_price HTTP error: 429 for COTIUSDT_SPBL
2025-02-07 20:51:43,546 - ERROR - Bitget get_price HTTP error: 429 for SYSUSDT_SPBL
2025-02-07 20:51:43,550 - ERROR - Bitget get_price HTTP error: 429 for APTUSDC_SPBL
2025-02-07 20:51:43,552 - ERROR - Bitget get_price HTTP error: 429 for WLDUSDT_SPBL
2025-02-07 20:51:43,561 - ERROR - Bitget get_price HTTP error: 429 for MBOXUSDT_SPBL
2025-02-07 20:51:43,580 - ERROR - Bitget get_price HTTP error: 429 for WUSDT_SPBL
2025-02-07 20:51:43,583 - ERROR - Bitget get_price HTTP error: 429 for RUNEUSDT_SPBL
2025-02-07 20:51:43,588 - ERROR - Bitget get_price HTTP error: 429 for JOEUSDT_SPBL
2025-02-07 20:51:43,619 - ERROR - Bitget get_price HTTP error: 429 for RDNTUSDT_SPBL
2025-02-07 20:51:43,625 - ERROR - Bitget get_price HTTP error: 429 for METISUSDT_SPBL
2025-02-07 20:51:43,627 - ERROR - Bitget get_price HTTP error: 429 for ATOMUSDT_SPBL
2025-02-07 20:51:44,269 - ERROR - Bitget get_price HTTP error: 429 for NEOUSDT_SPBL
2025-02-07 20:51:44,281 - ERROR - Bitget get_price HTTP error: 429 for BSWUSDT_SPBL
2025-02-07 20:51:44,290 - ERROR - Bitget get_price HTTP error: 429 for MOVEUSDT_SPBL
2025-02-07 20:51:44,321 - ERROR - Bitget get_price HTTP error: 429 for RADUSDT_SPBL
2025-02-07 20:51:44,325 - ERROR - Bitget get_price HTTP error: 429 for XRPETH_SPBL
2025-02-07 20:51:44,378 - ERROR - Bitget get_price HTTP error: 429 for EOSUSDT_SPBL
2025-02-07 20:51:44,388 - ERROR - Bitget get_price HTTP error: 429 for PSGUSDT_SPBL
2025-02-07 20:51:44,394 - ERROR - Bitget get_price HTTP error: 429 for SSVUSDT_SPBL
2025-02-07 20:51:44,398 - ERROR - Bitget get_price HTTP error: 429 for BLURUSDT_SPBL
2025-02-07 20:51:44,409 - ERROR - Bitget get_price HTTP error: 429 for VTHOUSDT_SPBL
2025-02-07 20:51:44,414 - ERROR - Bitget get_price HTTP error: 429 for EGLDUSDT_SPBL
2025-02-07 20:51:44,422 - ERROR - Bitget get_price HTTP error: 429 for PERPUSDT_SPBL
2025-02-07 20:51:44,445 - ERROR - Bitget get_price HTTP error: 429 for ACEUSDT_SPBL
2025-02-07 20:51:44,450 - ERROR - Bitget get_price HTTP error: 429 for APEUSDT_SPBL
2025-02-07 20:51:44,454 - ERROR - Bitget get_price HTTP error: 429 for IOSTUSDT_SPBL
2025-02-07 20:51:44,459 - ERROR - Bitget get_price HTTP error: 429 for TURBOUSDT_SPBL
2025-02-07 20:51:44,461 - ERROR - Bitget get_price HTTP error: 429 for SNXUSDT_SPBL
2025-02-07 20:51:44,466 - ERROR - Bitget get_price HTTP error: 429 for MEUSDT_SPBL
2025-02-07 20:51:44,473 - ERROR - Bitget get_price HTTP error: 429 for SUIEUR_SPBL
2025-02-07 20:51:44,475 - ERROR - Bitget get_price HTTP error: 429 for RLCUSDT_SPBL
2025-02-07 20:51:44,477 - ERROR - Bitget get_price HTTP error: 429 for DOGEUSDC_SPBL
2025-02-07 20:51:44,482 - ERROR - Bitget get_price HTTP error: 429 for LISTAUSDT_SPBL
2025-02-07 20:51:44,486 - ERROR - Bitget get_price HTTP error: 429 for BALUSDT_SPBL
2025-02-07 20:51:44,489 - ERROR - Bitget get_price HTTP error: 429 for STPTUSDT_SPBL
2025-02-07 20:51:44,491 - ERROR - Bitget get_price HTTP error: 429 for KAVAUSDT_SPBL
2025-02-07 20:51:44,493 - ERROR - Bitget get_price HTTP error: 429 for LUMIAUSDT_SPBL
2025-02-07 20:51:44,499 - ERROR - Bitget get_price HTTP error: 429 for API3USDT_SPBL
2025-02-07 20:51:44,504 - ERROR - Bitget get_price HTTP error: 429 for ASTRUSDT_SPBL
2025-02-07 20:51:44,509 - ERROR - Bitget get_price HTTP error: 429 for CETUSUSDT_SPBL
2025-02-07 20:51:44,512 - ERROR - Bitget get_price HTTP error: 429 for YFIUSDT_SPBL
2025-02-07 20:51:45,893 - ERROR - Kucoin get_price error for SWE-ATUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:45,900 - ERROR - Kucoin get_price error for ORD-IUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:45,922 - ERROR - Kucoin get_price error for KMN-OUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:45,937 - ERROR - Kucoin get_price error for OPU-SDC: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:45,948 - ERROR - Kucoin get_price error for SMI-LEUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:46,014 - ERROR - Bitget get_price HTTP error: 429 for BURGERUSDT_SPBL
2025-02-07 20:51:46,021 - ERROR - Bitget get_price HTTP error: 429 for BONDUSDT_SPBL
2025-02-07 20:51:46,025 - ERROR - Bitget get_price HTTP error: 429 for SILLYUSDT_SPBL
2025-02-07 20:51:46,027 - ERROR - Bitget get_price HTTP error: 429 for LOOMUSDT_SPBL
2025-02-07 20:51:46,033 - ERROR - Bitget get_price HTTP error: 429 for AVAUSDT_SPBL
2025-02-07 20:51:46,251 - ERROR - Kucoin get_price error for SLE-RFUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:46,577 - ERROR - Kucoin get_price error for FLO-WUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:46,593 - ERROR - Kucoin get_price error for PLU-MEUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:46,613 - ERROR - Kucoin get_price error for ANY-ONEUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:46,625 - ERROR - Kucoin get_price error for CLA-YUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:46,633 - ERROR - Kucoin get_price error for DOG-SUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:46,655 - ERROR - Kucoin get_price error for YUL-IUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:46,707 - ERROR - Kucoin get_price error for ELO-NUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:46,718 - ERROR - Kucoin get_price error for FUS-DT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:46,722 - ERROR - Kucoin get_price error for GRA-SSUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:46,745 - ERROR - Kucoin get_price error for SUS-DT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:46,786 - ERROR - Kucoin get_price error for SUN-DOGUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:46,799 - ERROR - Kucoin get_price error for FID-AUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:46,806 - ERROR - Kucoin get_price error for FUE-LUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:46,812 - ERROR - Kucoin get_price error for MAS-AUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:46,820 - ERROR - Kucoin get_price error for ORA-IUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:46,833 - ERROR - Kucoin get_price error for NTR-NUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:46,841 - ERROR - Kucoin get_price error for FLO-KIUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:46,848 - ERROR - Kucoin get_price error for LOO-MUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:46,861 - ERROR - Kucoin get_price error for MOO-DENGUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:46,871 - ERROR - Kucoin get_price error for AER-GOUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:46,876 - ERROR - Kucoin get_price error for SIL-LYUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:46,883 - ERROR - Kucoin get_price error for CAK-EUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:46,889 - ERROR - Kucoin get_price error for PAX-GUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:46,900 - ERROR - Kucoin get_price error for ORC-AUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:46,906 - ERROR - Kucoin get_price error for MIN-AUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:46,912 - ERROR - Kucoin get_price error for L3U-SDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:46,920 - ERROR - Kucoin get_price error for GTA-IUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:46,927 - ERROR - Kucoin get_price error for SDU-SDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:46,938 - ERROR - Kucoin get_price error for AVA-ILUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:46,955 - ERROR - Kucoin get_price error for REN-DERUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:46,963 - ERROR - Kucoin get_price error for USD-TBRL: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:46,968 - ERROR - Kucoin get_price error for LOG-XUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:46,979 - ERROR - Kucoin get_price error for TLO-SUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:47,050 - ERROR - Kucoin get_price error for COO-KIEUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:47,057 - ERROR - Kucoin get_price error for BEE-RUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:47,073 - ERROR - Kucoin get_price error for VAN-RYUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:47,140 - ERROR - Kucoin get_price error for GOA-TSUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:47,147 - ERROR - Kucoin get_price error for BON-DUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:47,153 - ERROR - Kucoin get_price error for HBA-RUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:47,170 - ERROR - Kucoin get_price error for BUR-GERUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:47,180 - ERROR - Kucoin get_price error for CRO-SUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:47,189 - ERROR - Kucoin get_price error for OSM-OUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:47,197 - ERROR - Kucoin get_price error for PUF-FERUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:47,212 - ERROR - Kucoin get_price error for XTE-RUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:47,222 - ERROR - Kucoin get_price error for MUB-IUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:47,290 - ERROR - Kucoin get_price error for GME-EUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:47,304 - ERROR - Kucoin get_price error for SAF-EUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:47,322 - ERROR - Kucoin get_price error for MAN-TAUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:47,340 - ERROR - Kucoin get_price error for HIP-POUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:47,356 - ERROR - Kucoin get_price error for BIG-TIMEUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:47,363 - ERROR - Kucoin get_price error for BLO-KUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:47,393 - ERROR - Kucoin get_price error for MIG-GLESUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:47,398 - ERROR - Kucoin get_price error for ANI-MEUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:47,404 - ERROR - Kucoin get_price error for MTO-SUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:47,414 - ERROR - Kucoin get_price error for DOD-OUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:47,433 - ERROR - Kucoin get_price error for ZER-CUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:47,440 - ERROR - Kucoin get_price error for NEI-ROUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:47,447 - ERROR - Kucoin get_price error for UNF-IUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:47,456 - ERROR - Kucoin get_price error for TOM-IUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:47,462 - ERROR - Kucoin get_price error for COM-BOUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:47,489 - ERROR - Kucoin get_price error for VAN-AUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:47,503 - ERROR - Kucoin get_price error for MEL-ANIAUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:47,514 - ERROR - Kucoin get_price error for MYR-IAUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:47,521 - ERROR - Kucoin get_price error for STR-KUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:47,529 - ERROR - Kucoin get_price error for MOR-PHOUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:47,538 - ERROR - Kucoin get_price error for USD-EUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:47,555 - ERROR - Kucoin get_price error for ALI-CEUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:47,561 - ERROR - Kucoin get_price error for OND-OUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:47,598 - ERROR - Kucoin get_price error for CGP-TUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:47,602 - ERROR - Kucoin get_price error for TUS-DUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:47,634 - ERROR - Kucoin get_price error for SUP-ERUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:47,655 - ERROR - Kucoin get_price error for PST-AKEUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:47,683 - ERROR - Kucoin get_price error for BRE-TTUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:47,688 - ERROR - Kucoin get_price error for HMS-TRUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:47,705 - ERROR - Kucoin get_price error for AAV-EBTC: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:47,716 - ERROR - Kucoin get_price error for FOR-TUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:47,729 - ERROR - Kucoin get_price error for ZKU-SDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:47,735 - ERROR - Kucoin get_price error for XIO-NUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:47,773 - ERROR - Kucoin get_price error for AIT-ECHUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:47,803 - ERROR - Kucoin get_price error for KOM-AUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:47,809 - ERROR - Kucoin get_price error for OMU-SDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:47,819 - ERROR - Kucoin get_price error for TNS-RUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:47,826 - ERROR - Kucoin get_price error for ETH-FIUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:47,835 - ERROR - Kucoin get_price error for GOA-TUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:47,844 - ERROR - Kucoin get_price error for CYB-ERUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:47,878 - ERROR - Kucoin get_price error for CPO-OLUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:47,883 - ERROR - Kucoin get_price error for BON-KUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:47,894 - ERROR - Kucoin get_price error for ZET-AUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:47,902 - ERROR - Kucoin get_price error for PEP-E2USDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:47,915 - ERROR - Kucoin get_price error for IOU-SDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:47,923 - ERROR - Kucoin get_price error for BUZ-ZUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:47,930 - ERROR - Kucoin get_price error for LUN-CUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:47,959 - ERROR - Kucoin get_price error for HOL-DUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:47,965 - ERROR - Kucoin get_price error for GRA-ILUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:47,972 - ERROR - Kucoin get_price error for NEI-ROCTOUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:47,981 - ERROR - Kucoin get_price error for LIN-KBTC: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:47,999 - ERROR - Kucoin get_price error for LIN-KUSDC: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:48,007 - ERROR - Bitget get_price HTTP error: 429 for DOGEUSDT_SPBL
2025-02-07 20:51:48,015 - ERROR - Kucoin get_price error for XCA-DUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:48,022 - ERROR - Bitget get_price HTTP error: 429 for VIRTUALUSDT_SPBL
2025-02-07 20:51:48,026 - ERROR - Kucoin get_price error for PON-KEUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:48,051 - ERROR - Kucoin get_price error for MAG-ICUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:48,061 - ERROR - Kucoin get_price error for BLA-STUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:48,101 - ERROR - Kucoin get_price error for ORD-ERUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:48,106 - ERROR - Kucoin get_price error for SON-ICUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:48,115 - ERROR - Kucoin get_price error for DUC-KUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:48,134 - ERROR - Bitget get_price HTTP error: 429 for SANDUSDT_SPBL
2025-02-07 20:51:48,137 - ERROR - Kucoin get_price error for DOG-EBTC: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:48,146 - ERROR - Bitget get_price HTTP error: 429 for DUSKUSDT_SPBL
2025-02-07 20:51:48,150 - ERROR - Kucoin get_price error for VIR-TUALUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:48,154 - ERROR - Kucoin get_price error for MOV-RUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:48,160 - ERROR - Bitget get_price HTTP error: 429 for CHZUSDT_SPBL
2025-02-07 20:51:48,170 - ERROR - Bitget get_price HTTP error: 429 for BICOUSDT_SPBL
2025-02-07 20:51:48,185 - ERROR - Bitget get_price HTTP error: 429 for CELRUSDT_SPBL
2025-02-07 20:51:48,193 - ERROR - Kucoin get_price error for DOG-EUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:48,198 - ERROR - Bitget get_price HTTP error: 429 for XCNUSDT_SPBL
2025-02-07 20:51:48,200 - ERROR - Kucoin get_price error for JAS-MYUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:48,209 - ERROR - Bitget get_price HTTP error: 429 for PEOPLEUSDT_SPBL
2025-02-07 20:51:48,212 - ERROR - Bitget get_price HTTP error: 429 for UNIOUSDT_SPBL
2025-02-07 20:51:48,215 - ERROR - Kucoin get_price error for FIR-EUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:48,229 - ERROR - Bitget get_price HTTP error: 429 for XRPUSDT_SPBL
2025-02-07 20:51:48,237 - ERROR - Kucoin get_price error for SAN-DUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:48,248 - ERROR - Kucoin get_price error for ZER-EBROUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:48,257 - ERROR - Kucoin get_price error for DUS-KUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:48,271 - ERROR - Kucoin get_price error for AEV-OUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:48,291 - ERROR - Kucoin get_price error for CEL-RUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:48,298 - ERROR - Kucoin get_price error for AI1-6ZUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:48,307 - ERROR - Kucoin get_price error for PUN-DIXUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:48,314 - ERROR - Bitget get_price HTTP error: 429 for FREDUSDT_SPBL
2025-02-07 20:51:48,316 - ERROR - Bitget get_price HTTP error: 429 for CFXUSDT_SPBL
2025-02-07 20:51:48,319 - ERROR - Kucoin get_price error for PYU-SDUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:48,326 - ERROR - Kucoin get_price error for ARU-SDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:48,332 - ERROR - Bitget get_price HTTP error: 429 for ILVUSDT_SPBL
2025-02-07 20:51:48,336 - ERROR - Kucoin get_price error for ARP-AUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:48,343 - ERROR - Kucoin get_price error for UNI-OUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:48,390 - ERROR - Bitget get_price HTTP error: 429 for LADYSUSDT_SPBL
2025-02-07 20:51:48,412 - ERROR - Kucoin get_price error for BIC-OUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:48,430 - ERROR - Bitget get_price HTTP error: 429 for STORJUSDT_SPBL
2025-02-07 20:51:48,438 - ERROR - Kucoin get_price error for PEO-PLEUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:48,448 - ERROR - Bitget get_price HTTP error: 429 for PRCLUSDT_SPBL
2025-02-07 20:51:48,455 - ERROR - Kucoin get_price error for ARC-AUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:48,479 - ERROR - Bitget get_price HTTP error: 429 for IDUSDT_SPBL
2025-02-07 20:51:48,483 - ERROR - Bitget get_price HTTP error: 429 for PUNDIXUSDT_SPBL
2025-02-07 20:51:48,496 - ERROR - Bitget get_price HTTP error: 429 for RACAUSDT_SPBL
2025-02-07 20:51:48,498 - ERROR - Kucoin get_price error for STO-RJUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:48,503 - ERROR - Kucoin get_price error for PRC-LUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:48,507 - ERROR - Bitget get_price HTTP error: 429 for LSKUSDT_SPBL
2025-02-07 20:51:48,510 - ERROR - Bitget get_price HTTP error: 429 for AVAXBTC_SPBL
2025-02-07 20:51:48,512 - ERROR - Bitget get_price HTTP error: 429 for BLZUSDT_SPBL
2025-02-07 20:51:48,514 - ERROR - Bitget get_price HTTP error: 429 for BTCEUR_SPBL
2025-02-07 20:51:48,517 - ERROR - Bitget get_price HTTP error: 429 for LUCEUSDT_SPBL
2025-02-07 20:51:48,521 - ERROR - Kucoin get_price error for AUD-IOUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:48,527 - ERROR - Kucoin get_price error for USD-CEUR: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:48,534 - ERROR - Bitget get_price HTTP error: 429 for XTZUSDT_SPBL
2025-02-07 20:51:48,536 - ERROR - Bitget get_price HTTP error: 429 for TLMUSDT_SPBL
2025-02-07 20:51:48,545 - ERROR - Kucoin get_price error for LAD-YSUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:48,552 - ERROR - Bitget get_price HTTP error: 429 for PYTHUSDT_SPBL
2025-02-07 20:51:48,554 - ERROR - Kucoin get_price error for RAC-AUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:48,561 - ERROR - Kucoin get_price error for IDU-SDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:48,578 - ERROR - Bitget get_price HTTP error: 429 for LOOKSUSDT_SPBL
2025-02-07 20:51:48,583 - ERROR - Kucoin get_price error for LUC-EUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:48,590 - ERROR - Kucoin get_price error for FRE-DUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:48,596 - ERROR - Kucoin get_price error for AVA-XBTC: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:48,600 - ERROR - Kucoin get_price error for SUS-HIUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:48,605 - ERROR - Kucoin get_price error for PYT-HUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:48,610 - ERROR - Kucoin get_price error for CAR-VUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:48,616 - ERROR - Bitget get_price HTTP error: 429 for CARVUSDT_SPBL
2025-02-07 20:51:48,618 - ERROR - Bitget get_price HTTP error: 429 for NKNUSDT_SPBL
2025-02-07 20:51:48,624 - ERROR - Kucoin get_price error for LOO-KSUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:48,631 - ERROR - Kucoin get_price error for LIN-AUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:48,638 - ERROR - Bitget get_price HTTP error: 429 for LINAUSDT_SPBL
2025-02-07 20:51:48,668 - ERROR - Kucoin get_price error for CEL-OUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
2025-02-07 20:51:48,677 - ERROR - Bitget get_price HTTP error: 429 for BTCUSDT_SPBL
2025-02-07 20:51:48,679 - ERROR - Bitget get_price HTTP error: 429 for SWARMSUSDT_SPBL
2025-02-07 20:51:48,681 - ERROR - Kucoin get_price error for SWA-RMSUSDT: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/workspaces/Arbitrage/exchanges/kucoin.py", line 52, in get_price
    price_str = data.get("data", {}).get("price", "0")
                ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
